### PR TITLE
Improve Baes anisotropy kernels and regularize LOS quadrature

### PR DESCRIPTION
**Pull request by gomeshun**
## Summary

Improve the numerical treatment of Baes–van Hese anisotropy in the JAX/NumPyro Jeans solver, motivated by an exact analytic result for the special case `eta = 2` and by matched-accuracy MCMC-oriented stress tests.

The PR now contains three related improvements:

1. an exact Appell-`F1` reduction of the `eta=2` anisotropy kernel,
2. a regularized outer LOS integration variable, `log(u) = x^2` (`sqrtlog`), and
3. a more stable log-domain implementation of the generic Baes kernel with a prior-backed default `n_kernel=32`.

## Exact `eta=2` kernel

For

```text
beta(r) = [beta_0 + beta_inf (r/r_a)^2] / [1 + (r/r_a)^2]
```

write

```text
p  = beta_0
q  = beta_inf - beta_0
a  = r_a / R
z1 = 1 - u^(-2)
z2 = (u^2 - 1) / (u^2 + a^2)
```

Then

```text
K = sqrt(z1) * [
    F1(1; p,   q;   3/2; z1, z2)
  - p/u^2       F1(1; p+1, q;   3/2; z1, z2)
  - q/(u^2+a^2) F1(1; p,   q+1; 3/2; z1, z2)
]
```

is an exact representation of the LOS anisotropy kernel. `BaesEta2AnisotropyModel` includes this result plus an independent high-precision `mpmath.appellf1` reference evaluator. Since JAX has no native Appell `F1`, the JAX hot path evaluates its Euler representation on a fixed Gauss–Legendre interval.

## Regularized outer LOS quadrature

The previous kernel solver used a uniform grid in `t = log(u)`. Near the lower endpoint, `K(u) ~ sqrt(u-1) ~ sqrt(t)`, which limits convergence.

The new default uses

```text
log(u) = x^2
u      = exp(x^2)
```

so both the kernel and Jacobian vanish smoothly at `x=0`. The legacy `kernel_outer_transform="log"` path remains available explicitly.

Matched-accuracy benchmarks show that this variable change, rather than the Appell evaluator itself, is the main runtime improvement for full `sigma_los^2` calculations.

## Generic Baes stability and kernel order

The generic kernel previously formed `(r/r_a)**eta` directly. This can overflow in float32 for sharp transitions even when the physical `beta(r)` and the required integrating-factor ratio remain finite. The new implementation uses

```text
ell = eta * log(r/r_a)
beta = beta_0 + (beta_inf-beta_0) * sigmoid(ell)
```

and evaluates the integrating-factor ratio through `softplus` differences in log space.

The default inner Gauss–Legendre order is reduced from 128 to 32. This is based on a 72-case deterministic stress set covering

- `eta = 0.1, 1, 4, 10`,
- anisotropy edges down to `beta=-9` and up to `beta=0.98`,
- `r_a/R_e = 0.005, 1, 50`, and
- `R/R_e = 0.005--10`.

With the outer grid held fixed and `n_kernel=256` as reference, the worst full-projection error from `n_kernel=32` was

- float64: `9.0e-8`,
- float32: `8.0e-6`.

`n_kernel=16` was noticeably worse (`~7.5e-4`), so 32 is retained as the practical lower default rather than reducing further.

## `eta` validation guidance

Recent dSph analyses using this Baes–van Hese parameterization repeatedly adopt `eta <= 10`, including:

- Hayashi, Ferreira & Chan (2021), FDM UFD analysis, `1 <= eta <= 10`: https://arxiv.org/abs/2102.05300
- Hayashi et al. (2021), SIDM UFD analysis, `1 <= eta <= 10`: https://doi.org/10.1103/PhysRevD.103.023017
- Ando, Hayashi, Horigome, Ibe & Shirai (2025), `1 <= eta <= 10`: https://arxiv.org/abs/2503.13650

More conservative Bonnivard-style Jeans analyses use an even narrower range (`eta` roughly `0.1--4`).

Accordingly, concrete inputs with `eta > 10` now emit a `RuntimeWarning`. The calculation is not rejected: the log-domain implementation remains usable outside the recommended range, but users are prompted to check convergence explicitly. JAX tracers do not trigger Python callbacks, so the warning does not enter an MCMC hot loop.

The `eta>10` adversarial tests remain useful: before the log-domain rewrite, float32 could return finite but catastrophically wrong results at `eta=20`; the stable implementation avoids that failure mode.

## Validation

Tests cover:

- exact `mpmath.appellf1` evaluation of the `eta=2` formula,
- generic numerical Baes kernel comparisons,
- constant-anisotropy and Osipkov–Merritt limits,
- kernel vs kernel-free Jeans + Abel `sigma_los^2`,
- `sqrtlog` vs legacy outer integration,
- adversarial radial/tangential and halo-scale stress cases,
- `eta>10` warning behavior,
- float32 finite behavior for `eta=20`, and
- `n_kernel=32` vs 128 at the supported-prior edge.

Latest clean CI:

- Python 3.12: 177 passed, 3 skipped
- Python 3.13: 177 passed, 3 skipped
- base-install smoke tests: passed on both Python versions
- benchmark steps: passed

## Scope / remaining work

This PR does **not** change the existing `auto` routing of generic Baes models to the Abel backend, and it does not yet retune the global `n_u` / `u_max` defaults. The supported-prior stress test shows that, once `n_kernel=32` is used, the dominant remaining error in aggressive cases comes from the outer/tail integration rather than the inner anisotropy-kernel quadrature. That tuning can therefore be handled separately.

### CODE REVIEW

**Comment by gomeshun** [comment]
## CPU benchmark results

Added `scripts/benchmark_baes_eta2.py` and measured the generic BAES kernel, the specialized `eta=2` Appell/Euler kernel, and the kernel-free Abel backend on the same GitHub Actions Ubuntu CPU runner.

Full benchmark configuration: 32 projected radii, `n_u=256`, `n_kernel=128`, `n_r_abel=768`, 7 timed repeats after warmup. Plummer tracer + NFW halo with `(beta_0, beta_inf, r_a)=(-0.5, 0.65, 300 pc)`.

| mode | generic kernel | eta=2 kernel | kernel speedup | generic sigma_los^2 | eta=2 sigma_los^2 | Abel sigma_los^2 | eta2/generic speedup |
|---|---:|---:|---:|---:|---:|---:|---:|
| CPU float32 | 16.09 ms | 4.49 ms | **3.58x** | 5.69 ms | 1.70 ms | 0.379 ms | **3.34x** |
| CPU float64 | 46.09 ms | 19.27 ms | **2.39x** | 15.70 ms | 6.73 ms | 0.768 ms | **2.33x** |

First-call/JIT costs also improve: for `sigma_los^2`, generic vs eta=2 was 0.430 s vs 0.255 s (float32), and 0.389 s vs 0.257 s (float64).

Accuracy in this run:

- eta=2 vs generic kernel max relative difference: `7.44e-5` (float32), `1.81e-13` (float64)
- eta=2 vs generic `sigma_los^2`: `1.93e-7` (float32), `5.39e-14` (float64)
- eta=2 kernel vs Abel `sigma_los^2`: about `3.34e-3` in both precisions, consistent with the different integration resolutions rather than precision loss

The analytic reduction therefore gives a clear 2–4x speedup over the current generic BAES kernel. However, for this particular model/grid the existing kernel-free Abel backend is still substantially faster (about 4.5x faster than eta=2 in float32 and 8.8x in float64 at these resolutions).

The next meaningful benchmark should therefore compare **runtime at matched numerical accuracy**, scanning `n_kernel`, `n_u`, and `n_r` rather than fixing nominal resolutions. That will show whether the smoother eta=2 fixed-interval integral wins at a target error tolerance, even though Abel is faster at the current settings.

The benchmark was temporarily run in CI to obtain a reproducible shared-runner measurement; the CI benchmark step has now been removed, while the reusable benchmark script remains in this PR.

**Comment by gomeshun** [comment]
## Matched-accuracy benchmark

Added `scripts/benchmark_baes_eta2_accuracy.py` and compared the three `sigma_los^2` routes at matched numerical accuracy on the same GitHub Actions Ubuntu CPU runner (JAX float64, 32 projected radii, `u_max=5000`).

Two independent high-resolution references were used:

- eta=2 kernel: `n_u=8192`, `n_kernel=64`
- Abel backend: `n_r=32768`

Their residual difference is `1.65e-5` max relative (`7.25e-6` RMS), so the sweep is reliable down to roughly the `1e-4` level.

| max relative error target | generic Baes | eta=2 analytic kernel | Abel | eta=2 speedup vs generic | Abel speedup vs eta=2 |
|---|---:|---:|---:|---:|---:|
| `1e-2` | 2.266 ms | 1.016 ms | 0.280 ms | 2.23x | 3.62x |
| `3e-3` | 4.436 ms | 1.926 ms | 0.562 ms | 2.30x | 3.43x |
| `1e-3` | 8.740 ms | 3.723 ms | 0.963 ms | 2.35x | 3.87x |
| `3e-4` | 33.511 ms | 14.362 ms | 1.813 ms | 2.33x | 7.92x |
| `1e-4` | 70.389 ms | 29.103 ms | 4.240 ms | 2.42x | 6.86x |

At `3e-5`, the scanned Abel configuration (`n_r=16384`) still reaches the tolerance in 8.64 ms; the scanned kernel grids do not yet reach that tolerance.

### Important numerical observation

For the eta=2 route at fixed `n_u=512`, changing the inner kernel quadrature from `n_kernel=32` to 64 or 128 changes the final error only at the ~`4e-13` level:

```text
n_kernel=32   error=9.4589663176e-4
n_kernel=64   error=9.4589663217e-4
n_kernel=128  error=9.4589663217e-4
```

So the eta=2 kernel quadrature is already effectively converged at `n_kernel=32`; the dominant error/cost is the **outer `u` integration**, not the Appell/Euler inner integral. This also means that replacing the current Euler quadrature with a faster native `F1` evaluator would improve the kernel call itself, but is unlikely by itself to close the large gap to the Abel backend for full `sigma_los^2` evaluation.

### Interpretation

The analytic eta=2 kernel consistently gives a useful ~2.3–2.4x speedup over the generic Baes kernel at fixed target accuracy. However, the current kernel-free Abel backend remains clearly faster at every tested accuracy, by roughly 3.4–7.9x relative to eta=2.

Therefore, if the primary objective is raw `sigma_los^2` inference speed, the next optimization target should be the **outer kernel integration / overall projection strategy**, not the inner anisotropy-kernel evaluator. The analytic kernel remains valuable as an exact result, a validation route, and for applications that require the kernel itself.

The benchmark step was run temporarily in CI for reproducibility and has been removed again; the reusable benchmark script remains in the PR.

File: src/jeanspy/baes_eta2.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+from .model_numpyro import BaesAnisotropyModel
+
+import jax
+import jax.numpy as jnp
+
+
+DEFAULT_BAES_ETA2_KERNEL_N_QUAD = 96
+
+
+@lru_cache(maxsize=None)
```
[the remark this answers is not in the document]
Comment (gomeshun) [reply src/jeanspy/baes_eta2.py]: Fixed in 8907344 by bounding the eta=2 Gauss–Legendre cache at maxsize=16, matching model_numpyro._gauss_legendre_01. This keeps the small set of commonly used quadrature orders cached without unbounded growth.
